operator-framework/src/test/java/io/javaoperatorsdk/operator/sample/perresourceeventsource/PerResourcePollingEventSourceTestReconciler.java

```diff
@@ -0,0 +1,65 @@
+package io.javaoperatorsdk.operator.sample.perresourceeventsource;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.javaoperatorsdk.operator.api.reconciler.*;
+import io.javaoperatorsdk.operator.junit.KubernetesClientAware;
+import io.javaoperatorsdk.operator.processing.event.source.EventSource;
+import io.javaoperatorsdk.operator.processing.event.source.polling.PerResourcePollingEventSource;
+
+@ControllerConfiguration
+public class PerResourcePollingEventSourceTestReconciler
+    implements Reconciler<PerResourceEventSourceCustomResource>,
+    EventSourceInitializer<PerResourceEventSourceCustomResource>,
+    KubernetesClientAware {
+
+  public static final int POLL_PERIOD = 100;
+  private final AtomicInteger numberOfExecutions = new AtomicInteger(0);
+  private final AtomicInteger numberOfFetchExecutions = new AtomicInteger(0);
+
+  private KubernetesClient client;
+  private PerResourcePollingEventSource<String, PerResourceEventSourceCustomResource> eventSource;
+
+  @Override
+  public UpdateControl<PerResourceEventSourceCustomResource> reconcile(
+      PerResourceEventSourceCustomResource resource,
+      Context<PerResourceEventSourceCustomResource> context) throws Exception {
+    numberOfExecutions.addAndGet(1);
+    return UpdateControl.noUpdate();
+  }
+
+  @Override
+  public Map<String, EventSource> prepareEventSources(
+      EventSourceContext<PerResourceEventSourceCustomResource> context) {
+    this.eventSource =
+        new PerResourcePollingEventSource<>(primaryResource -> {
+          numberOfFetchExecutions.addAndGet(1);
```

Comment:
Wouldn't the test make more sense if it checked that it's actually called once per primary resource? Right now, it only checks the fetcher is called, not that it's called per resource.

Reply:
Yes, true, will improve the test.

Reply:
updated